Sources/eppo/AssignmentCache.swift

```diff
@@ -13,6 +13,7 @@ public struct AssignmentCacheKey {
 }
 
 public class InMemoryAssignmentCache: AssignmentCache {
+    private let queue = DispatchQueue(label: "com.eppo.assignmentcache", attributes: .concurrent)
```

Comment:
minor: have you considered using a Mutex instead? Given our sync blocks are very short, I would expect it to be a better fit here

Reply:
I did give this a read - https://swiftrocks.com/thread-safety-in-swift

Does seem like an `NSLock` is the fastest but I value my understanding of this API and feel comfortable with it working, but would be open to re-implementing the usage here now that we have a test that demonstrates the bug.